File: OneDrive/Documents/Unam}/Ciclo-III/Analisis y diseños de algoritmos/Informe/quicksort.cpp
#include <iostream>
using namespace std;
int particion(int arreglo[], int izquierda, int derecha)
{
    int pivote = arreglo[izquierda];
    while (true)
    {
        while (arreglo[izquierda] < pivote)
            izquierda++;
        while (arreglo[derecha] > pivote)
            derecha--;
        if (izquierda >= derecha)
            return derecha;
        else
        {
            int temp = arreglo[izquierda];
            arreglo[izquierda] = arreglo[derecha];
            arreglo[derecha] = temp;
            izquierda++;
            derecha--;
        }
    }
}

void quicksort(int arreglo[], int izquierda, int derecha)
{
    if (izquierda < derecha)
    {
        int indiceParticion = particion(arreglo, izquierda, derecha);
        quicksort(arreglo, izquierda, indiceParticion);
        quicksort(arreglo, indiceParticion + 1, derecha);
}
}

File: OneDrive/Documents/Unam}/Ciclo-III/Analisis y diseños de algoritmos/Informe/QuickSort.py


def particion(arreglo, izquierda, derecha):
    pivote = arreglo[izquierda]
    while True:
        while arreglo[izquierda] < pivote:
            izquierda += 1
        while arreglo[derecha] > pivote:
            derecha -= 1
        if izquierda >= derecha:
            return derecha
        else:
            arreglo[izquierda], arreglo[derecha] = arreglo[derecha], arreglo[izquierda]
            izquierda += 1
            derecha -= 1
def quicksort(arreglo,izquierda,derecha):
    if izquierda < derecha:
        indiceParticion = particion(arreglo, izquierda, derecha)
        quicksort(arreglo, izquierda, indiceParticion)
        quicksort(arreglo, indiceParticion + 1, derecha)
    return lista



